src/main/java/com/spotify/docker/client/messages/ContainerConfig.java

```diff
@@ -265,14 +265,7 @@ public static Builder builder() {
 
     public abstract Builder volumes(final Set<String> volumes);
 
-    public Builder volumes(final String... volumes) {
-      if (volumes != null) {
-        for (final String volume : volumes) {
-          addVolume(volume);
-        }
-      }
-      return this;
-    }
+    public abstract Builder volumes(final String... volumes);
```

Comment:
Just wanted to point out that the behavior is different here. This replaces whereas the previous version was cumulative. I prefer this one, but it would be backwards incompatible. We should point this out in the changelog if we merge this.

Reply:
No, this is the same. The old code had a method with the same signature which also replaced the volumes.

This method has been removed, but had the same signature as the new abstract method. I can show that it did replace and not accumulate. It was explicitly implemented to turned the varargs into a `Set` and pass it to another method:
```java
/**		 
 * @deprecated  As of release 7.0.0, replaced by {@link #volumes(Map)}.
 */		
@Deprecated		
public Builder volumes(final String... volumes) {		
  if (volumes != null && volumes.length > 0) {		
    volumes(ImmutableSet.copyOf(volumes));		
  }		
  return this;
}
```
Just from that we can't tell if it replaced or accumulated. But we can look at that second Builder method, the one that took a `Set` and turned it into a `Map<String, Map>`:
```java
/**
 * @deprecated  As of release 7.0.0, replaced by {@link #volumes(Map)}.		
 */		
@Deprecated		
public Builder volumes(final Set<String> volumes) {		
  if (volumes != null && !volumes.isEmpty()) {		
    final ImmutableMap.Builder<String, Map> volumesBuilder = ImmutableMap.builder();		
    for (final String volume : volumes) {		
      volumesBuilder.put(volume, new HashMap());		
    }		
    volumes(volumesBuilder.build());		
  }		
  return this;		
}
```

This passes the `Map<String, Map>` into the abstract Builder method that replaces the volumes. So the whole chain, from `volumes(String... volumes)`, all were replacing any existing contents.

The new method does the same.

Reply:
There was a method that accumulated: `addVolumes(String...)`. That one got removed. I'll add it back.

I'll also add back what had been the primary `Builder` method for `volumes`: `volumes(Map<String, Map>)`. I removed that, but I should have deprecated it.